homeassistant/components/homekit/type_switches.py

```diff
@@ -226,3 +233,53 @@ def async_update_state(self, new_state):
         self.char_active.set_value(current_state)
         _LOGGER.debug("%s: Set in_use state to %s", self.entity_id, current_state)
         self.char_in_use.set_value(current_state)
+
+
+@TYPES.register("SelectSwitch")
+class SelectSwitch(HomeAccessory):
+    """Generate a Switch accessory that contains multiple switches."""
+
+    def __init__(self, *args):
+        """Initialize a Switch accessory object."""
+        super().__init__(*args, category=CATEGORY_SWITCH)
+
+        state = self.hass.states.get(self.entity_id)
+        self.select_chars = {}
+        options = state.attributes.get("options", [])
+        if options:
```

Comment:
We probably shouldn't create the accessory if there are no options present (check in `accessories.py`)

Reply:
It seems that HASS doesn't allow an input_select with no options.

```
required key not provided @ data['options']. Got None
```

Reply:
Then we can do `state.attributes["options"]` instead